### ISSUE TITLE
Quantitative result on OASIS dataset

### ISSUE THREAD

**Issue by Valeyards** [xi-jia/LKU-Net#2]
Hi @xi-jia, thank you so much for sharing your work and code! Here I have a question: when I'm trying to reproduce the results of LKU-Net on OASIS dataset, the dice metric on validation set converges at around 80. I'm currently using 8-channels version. During training I adopted the same strategy on Transmorph which is randomly select a pair of image and feed them twice as fixed and moving image with dice loss. I wonder if there is something that I missed.

Best regards.

**Comment by xi-jia** [comment]
Hi @Valeyards , thank you so much for being interested in our work. My experience are 1) enlarging the network brings better performance. You are using the 8-channels version, switching into 16 and 32 (we used c=32, k=5) will potentially improve the dice score. 2) using segmentation masks brings better performance, we used dice loss + ncc / mse loss + regularization loss for all experiments. 3) The submission of L2R 2021 is required to be half-resolution displacements, one can train a full-resolution displacement first then downsample, or can train the half-resolution displacements directly. We notice a tiny improvement using the latter approach over the former.

**Comment by Valeyards** [comment]
> Hi @Valeyards , thank you so much for being interested in our work. My experience are 1) enlarging the network brings better performance. You are using the 8-channels version, switching into 16 and 32 (we used c=32, k=5) will potentially improve the dice score. 2) using segmentation masks brings better performance, we used dice loss + ncc / mse loss + regularization loss for all experiments. 3) The submission of L2R 2021 is required to be half-resolution displacements, one can train a full-resolution displacement first then downsample, or can train the half-resolution displacements directly. We notice a tiny improvement using the latter approach over the former.

Thx for your reply and congrats on your result in learn2reg 2022! I may have a try on your suggestions though I think numbers of channels may not affect the final performance in such a large margin. I thought my implementation might be kind of wrong. Do you have any plan on releasing OASIS version code?

**Comment by xi-jia** [comment]
> > Hi @Valeyards , thank you so much for being interested in our work. My experience are 1) enlarging the network brings better performance. You are using the 8-channels version, switching into 16 and 32 (we used c=32, k=5) will potentially improve the dice score. 2) using segmentation masks brings better performance, we used dice loss + ncc / mse loss + regularization loss for all experiments. 3) The submission of L2R 2021 is required to be half-resolution displacements, one can train a full-resolution displacement first then downsample, or can train the half-resolution displacements directly. We notice a tiny improvement using the latter approach over the former.
> 
> Thx for your reply and congrats on your result in learn2reg 2022! I may have a try on your suggestions though I think numbers of channels may not affect the final performance in such a large margin. I thought my implementation might be kind of wrong. Do you have any plan on releasing OASIS version code?

Sorry for the late reply, have you reproduced the results? I will update the OASIS training code and release trained models soon.

**Comment by Valeyards** [comment]
> > > Hi @Valeyards , thank you so much for being interested in our work. My experience are 1) enlarging the network brings better performance. You are using the 8-channels version, switching into 16 and 32 (we used c=32, k=5) will potentially improve the dice score. 2) using segmentation masks brings better performance, we used dice loss + ncc / mse loss + regularization loss for all experiments. 3) The submission of L2R 2021 is required to be half-resolution displacements, one can train a full-resolution displacement first then downsample, or can train the half-resolution displacements directly. We notice a tiny improvement using the latter approach over the former.
> > 
> > 
> > Thx for your reply and congrats on your result in learn2reg 2022! I may have a try on your suggestions though I think numbers of channels may not affect the final performance in such a large margin. I thought my implementation might be kind of wrong. Do you have any plan on releasing OASIS version code?
> 
> Sorry for the late reply, have you reproduced the results? I will update the OASIS training code and release trained models soon.

Thanks for your reply, I didn't make it reproducing the results. Look forward to your updates. :)

**Comment by xi-jia** [comment]
It is updated. The pre-trained models are also released and can be found in the readme.

**Comment by Valeyards** [comment]
Thx!

**Comment by windandink** [comment]
> Hi [@Valeyards](https://github.com/Valeyards) , thank you so much for being interested in our work. My experience are 1) enlarging the network brings better performance. You are using the 8-channels version, switching into 16 and 32 (we used c=32, k=5) will potentially improve the dice score. 2) using segmentation masks brings better performance, we used dice loss + ncc / mse loss + regularization loss for all experiments. 3) The submission of L2R 2021 is required to be half-resolution displacements, one can train a full-resolution displacement first then downsample, or can train the half-resolution displacements directly. We notice a tiny improvement using the latter approach over the former.

你好，我使用全分辨率的模型训练，然后提交下采样后的变形场，进行验证时发现精度相比全分辨率的结果下降了很多，dice系数下降了大约5%，请问您是否有遇到类似问题？

**Comment by xi-jia** [comment]
@windandink 是的，您引用的部分也写了：We notice a tiny improvement using the latter approach over the former。直接训练half-resolution提交会高些；先full再down会有些下降，但我记得没有5%这么多，有没有可能是不同采样的实现方式导致的，比如align corner，nearest/linear interpolation等等。
